File: Backend/src/main/java/de/ude/backend/model/questions/MultipleChoiceQuestion.java
package de.ude.backend.model.questions;

import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;
import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@Document(collection = "Questions")
@TypeAlias("MultipleChoiceQuestion")
public class MultipleChoiceQuestion extends Question {

    @NonNull
    private String[] choices;

    public MultipleChoiceQuestion(@NonNull String questionText, @NonNull String[] choices) {
        super(questionText);
        this.choices = choices;
    }

}

File: Extension/src/pages/popup/model/question/MultipleChoiceQuestion.ts
import {Question} from "@pages/popup/model/question/Question";
import {IMultipleChoiceQuestion} from "@pages/popup/Database/DatabaseInterfaces";

export class MultipleChoiceQuestion extends Question {

    private _choices: string[];

    constructor(id = "-1", question = "error", choices = ["error"]) {
        super(id, question, "MultipleChoiceQuestion");
        this._choices = choices;
    }

    get choices(): string[] {
        return this._choices;
    }

    set choices(value: string[]) {
        this._choices = value;
    }

    mapToIQuestion(): IMultipleChoiceQuestion {
        return {
            questionId: this.questionId,
            type: this.type,
            questionText: this.questionText,
            choices: this.choices
        };
    }
}